### DISCUSSION TITLE
has-text inside upward?

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uBlock-issues#2410] [Q&A] [answered]
`##.attention-contents:upward(.attention:has-text(وبگردی))`

Shouldn't this be possible?

On:
`https://www.entekhab.ir/fa/news/707058/%D9%88%D8%B2%DB%8C%D8%B1-%DA%A9%D8%B4%D9%88%D8%B1-%D9%81%D8%B6%D8%A7%DB%8C-%D9%85%D8%AC%D8%A7%D8%B2%DB%8C-%D8%A8%D9%87-%D8%B7%D9%88%D8%B1-%DA%A9%D8%A7%D9%85%D9%84-%D9%81%DB%8C%D9%84%D8%AA%D8%B1-%D9%85%DB%8C%E2%80%8C%D8%B4%D9%88%D8%AF-%D9%87%D8%B1%DA%86%D9%86%D8%AF-%D8%AF%D8%B4%D9%85%D9%86%D8%A7%D9%86-%D8%AF%D8%B1-%D8%AF%D8%B3%D8%AA%D9%88%D8%B1-%DA%A9%D8%A7%D8%B1-%D8%AE%D9%88%D8%AF-%D8%B3%D8%A7%D8%AE%D8%AA-%D9%81%DB%8C%D9%84%D8%AA%D8%B1%D8%B4%DA%A9%D9%86%E2%80%8C%D9%87%D8%A7-%D8%B1%D8%A7-%D8%AF%D8%A7%D8%B1%D9%86%D8%AF`

to only hide the "وبگردی" section on the left.

**Comment by mapx-** [comment]
You can use

`entekhab.ir##:xpath("//*[contains(text(),'وبگردی')]"):upward(.attention)`

**Comment by gorhill** [comment, marked as answer]
> Shouldn't this be possible?

No, the [documentation says](https://github.com/gorhill/uBlock/wiki/Procedural-cosmetic-filters#subjectupwardarg) the arg to `upward()` must be an integer or a plain CSS selector.

Use:

    ##.attention > header:has-text(وبگردی):upward(.attention)

**Comment by u-RraaLL** [comment]
> `##.attention-contents:upward(.attention:has-text(وبگردی))`

And what's wrong with:

    ##.attention-contents:upward(.attention):has-text(وبگردی)

?

Actually, why not just:

    ##.attention:has-text(وبگردی)

in the first place?

**Reply by MasterKia** [reply]
> Actually, why not just: `##.attention:has-text(وبگردی)`

Would it be more efficient than
`##.attention > header:has-text(وبگردی):upward(.attention)`
?

Since your filter needs to look through _all_ the texts inside that div before finding a match.

**Reply by Yuki2718** [reply]
`##.attention` has only 2 matches so I won't care too much. You can do `#?#.attention:has(> header:has-text(وبگردی))` too tho I prefer `##.attention > header:has-text(وبگردی):upward(.attention)` in this case. But apparently `.attention-contents` does not includes `وبگردی`??
